### ISSUE TITLE
Server is not publishing user properties to lookup-server

### ISSUE THREAD

**Issue by abangtor** [nextcloud/server#25290]
There is a bug in the getUserAccountData method in the RetryJob.php of the lookup-server-connector module.
The properties are written into an one dimensional array $publicData:
https://github.com/nextcloud/server/blob/7668c861161553dff6b4ea6ced9d575b15a1fd3f/apps/lookup_server_connector/lib/BackgroundJobs/RetryJob.php#L197
Later on the array is read as it is a two dimensional array:
https://github.com/nextcloud/server/blob/7668c861161553dff6b4ea6ced9d575b15a1fd3f/apps/lookup_server_connector/lib/BackgroundJobs/RetryJob.php#L203-L214

This causes the values to be empty and no properties are send to the loopup-server.

I don't know how to fix this exact, since I couldn't find a way to get the signature value from the Account or AccountProperty classes.
My current *ugly* fix is: `$publicData[$property->getName()]['value'] = $property->getValue();`

**Comment by kesselb** [comment]
Mind to sent a pull request? I would ignore the signature field for now. Hopefully one of the reviewer will know ;) 

> $data['name'] = $publicData[IAccountManager::PROPERTY_DISPLAYNAME]['value'] ?? ''; 

I guess we need to remove ['value'] here to make it work.

**Comment by faust64** [comment]
So... Digging on that topic, we can find the previous copy of that code, somewhere around there:

https://github.com/nextcloud/server/commit/30051e786b7b39b95eef5a878836419125afb6df#diff-eb7b5fbd99fede3d2293d9e58f6e38f93f6e13d8240b304a72004665aec8d5ab

Sounds like @kesselb is right, we should be able to just remove the `['value']` on most of those.

Which leaves us with a couple `verified`. Sounds like we can call some `getVerified()` method on mail and twitter properties (accounts properties / IAccountProperty)
 https://github.com/nextcloud/server/blob/905e1918d2796b9a79025283cd6edf2c40f49d77/lib/public/Accounts/IAccountProperty.php

And a couple `signature` I'm not sure what to do with.
We can find other occurrences over there:
https://github.com/nextcloud/server/blob/d31435d7f554ea3c9d45918a018e431ac08fe612/apps/settings/lib/Controller/UsersController.php#L502

Though iAccountProperty and AccountManager won't mention signatures ... not sure I'm looking at the right place. I'll start with blank values for now.

**Comment by abangtor** [comment]
Yes, I taught so too, that's why I added them as todo in the pull request.
But my pull request failed the *Lint / php-cs check* (I guess because of the comment...).
Should I close my pull request, since your implementation looks more feature complete with the `verified` fields?

**Comment by kesselb** [comment]
Nah. I made a bad suggestion ;)

```
$publicData[$property->getName()] = [
    'value' => $property->getValue(),
    'verified' => $property->getVerified()
];
```

**Comment by abangtor** [comment]
Ok, changed my commit now to the example from @kesselb.
With the verified status.

**Comment by szaimen** [comment]
Hi, please update to 24.0.9 or better 25.0.3 and report back if it fixes the issue. Thank you!

My goal is to add a label like e.g. 25-feedback to this ticket of an up-to-date major Nextcloud version where the bug could be reproduced. However this is not going to work without your help. So thanks for all your effort!

If you don't manage to reproduce the issue in time and the issue gets closed but you can reproduce the issue afterwards, feel free to create a new bug report with up-to-date information by following this link: https://github.com/nextcloud/server/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=BUG_REPORT.yml&title=%5BBug%5D%3A+

**Comment by fcharlaix-opendsi** [comment]
@szaimen I can confirm this issue is still present in Nextcloud 25 (25.0.6), even in 26 because this file hasn't been updated since Aug 8 2022.

The PR #25300 fix this issue, you should review it !
